packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -29,6 +29,7 @@ export default createRule({
   meta: {
     docs: {
       description: 'Enforce consistent awaiting of returned promises',
+      recommended: 'strict',
```

Comment:
@kirkwaiblinger surfacing what you mentioned a bit back in a DM: that we might want to customize settings for this rule in the strict configs. I'm interpreting https://github.com/typescript-eslint/typescript-eslint/issues/8667#issuecomment-2148569319 to mean that:

* For `strict`: we want `in-type-checked`, as it avoids stylistic things
* For `stylistic`: that'd be a separate issue

...and since this already has `defaultOptions: ['in-try-catch']`, we're cleared to just say `recommended: 'strict'`. Is that right?

Reply:
negative - my understanding was that the only thing that we are currently proceeding with is putting `return-await` in the `strict` _with option `error-handling-correctness-only`_.

Reply:
Right! Makes sense, thanks.

Since that's a little bit more plumbing, as you suggested it would be, I'm going to close this one out & let you take lead. You've already taken the lead on so much around it & other async things, I'd rather not butt in.